app/helpers/topical_events_helper.rb

```diff
@@ -0,0 +1,11 @@
+module TopicalEventsHelper
+  def query_string_from_content_type(content_type, slug)
+    string_hash = {
+      publication: "/search/all?topical_events%5B%5D=",
+      consultation: "/search/policy-papers-and-consultations?content_store_document_type%5B%5D=open_consultations&content_store_document_type%5B%5D=closed_consultations&topical_events=",
+      announcement: "/search/news-and-communications?topical_events=",
+    }
+
+    string_hash[content_type] + slug
+  end
```

Comment:
Generating the query string will probably be clearer if you create an array of parameters then use the `to_query` method to turn this into a query string, e.g.

```ruby
def search_url(content_type, slug)
  base_path(content_type) + query_params(content_type, slug).to_query
end

private

def base_path(content_type)
  case content_type
  when :publication
    "/search/all"
  when :consultations
    "/search/policy-papers-and-consultations"
  when :announcement
    "/search/news-and-communications"
  end
end

def query_params(content_type, slug)
  case content_type
  when :publication
    [
      topical_events: [slug],
    ]
  when :consultations
    [
      content_store_document_type: ["open_consultations", "closed_consultations"],
      topical_events: [slug],
    ]
  when :announcement
    [
      topical_events: [slug],
    ]
  end
end
```

Reply:
Yup really like this, have copied almost verbatim in what I'm about to push up :-)